File: syncProxy/src/main/java/com/wfm/jdkproxy/Student.java
package com.wfm.jdkproxy;

/**
 * @Auther: wangfanming
 * @Date: 2019/7/17 16:27
 * @Description:
 */
public class Student implements People {


    public String work() {
        System.out.println("我在学习。。。");
        return "学生要学习";
    }
}

File: scalaStudy/src/main/scala/com/wfm/extend/Student.scala
package com.wfm.extend

/**
 *
 * @ClassName Student
 * @Descripyion TODO
 * @author wangfanming
 * @date 2020/2/25 17:18
 * @version 1.0
 */
class Student private (var name:String,var age:Int) {
    //私有主构造器，别的类不能直接使用，必须依赖其他辅助构造器
}
